blog/2022-06-03-LFX-Blog-Prashant.md

```diff
@@ -0,0 +1,107 @@
+          echo "VER=$VERSION" >> $GITHUB_ENV
+      - name: Bump new version in TOML files
+        run: |
+          OLD_VERSION=$(cargo pkgid | cut -d# -f2 | cut -d: -f2)
+          echo "OLD=$OLD_VERSION" >> $GITHUB_ENV
+          cargo set-version --workspace --bump ${{ env.VER }}
+          NEW_VERSION=$(cargo pkgid | cut -d# -f2 | cut -d: -f2)
+          echo "NEW=$NEW_VERSION" >> $GITHUB_ENV   
+```
+3. Commit, push, and Pull Request is created automatically with `Release` tag for the release. From there, the maintainers will do all the necessary reviews, and merge once the CI passes.
+
+### Publishing Release
+
+- So, the Draft Release pull request is merged, great! It automatically triggers the release workflow, which by the way only works if the PR has the `Release` tag, and ignores all other. This is achieved using the conditional statement:
+
+```yaml
+ if: github.event.pull_request.merged && contains( github.event.pull_request.labels.*.name, 'Release')
+```
+
+- Changelog is automatically extracted using this [great workflow action](https://github.com/ffurrer2/extract-release-notes), and the release is made.
+
+```yaml
+      - name: Extract release notes
+        id: extract-release-notes
+        uses: ffurrer2/extract-release-notes@v1
+      - name: Create release
+        uses: actions/create-release@v1
+```
+
+- To trigger the publish crates workflow, I used this [workflow dispatch action](https://github.com/benc-uk/workflow-dispatch) as shown below:
+
+```yaml
+      - name: Trigger publish crates workflow
+        uses: benc-uk/workflow-dispatch@v1
+        with:
+          workflow: Publish crates
+          token: ${{ secrets.PAT_TOKEN }}
+```
+And that's it for the release!
+
+### Publishing crates
+
+The Publish crates workflow is now triggered as mentioned in the previous state. There are 4 main crates to be published, and one job to trigger the draft release workflow for tremor-language-server repo (All automated!). Github actions makes it really great to see which job is interconnected.
+
+![image](https://user-images.githubusercontent.com/23097199/171507584-c8a1bb81-0cbc-4a70-966e-b998758e6430.png)
+
+With all the crates published, including the language-server which follows the exact same process. Tremor has successfully released a new version! Congratulations!
+
+## My thoughts
+
+The tremor community has been extremely helpful in guiding me through the entire mentorship. They have this principle of "Never *worry*, have fun" that will always stay forever with me, and forward in my career. Special mention for [Heinz](https://github.com/Licenser) who mentored me throughout the months and helped me. And to the tremor community in general, my thanks to all of them! 
+I didn't know much about github actions or DevOps in general. But now I can confidently say that I can indeed, _make processes boring by automating them_. I will continue to engage in open source projects, and guide others to the same, cheers! 
```

Comment:
Did you mean less boring? :D

Reply:
Oh, wow. What a nice blog post! Good work, @PrimalPimmy! :)

Reply:
I think boring is exactly the right word here  :D, boring in a good way!

Reply:
Nope, I'm quoting @darach :p

Reply:
Oh, I didn't get that at all! :D

Reply:
> Oh, wow. What a nice blog post! Good work, @PrimalPimmy! :)

Thanks mate! :rocket: